nucliadb/nucliadb/ingest/orm/processor.py

```diff
@@ -467,6 +467,44 @@ async def txn(
             counter=counter,
         )
 
+    async def _mark_resource_error(
+        self,
+        resource: Optional[Resource],
+        partition: str,
+        seqid: int,
+        shard: Optional[Shard],
+        kbid: str,
+    ) -> None:
+        """
+        Unhandled error processing, try to mark resource as error
+        """
+        if shard is None:
+            logger.warning(
+                "Unable to mark resource as error, shard is None. "
+                "This should not happen so you did something special to get here."
+            )
+            return
+        if resource is None or resource.basic is None:
+            logger.info(
+                f"Skip when resource does not even have basic metadata: {resource}"
+            )
+            return
+        txn = None
```

Comment:
FYW you can also use the `with self.driver.transaction() as txn:` context manager that will handle aborting for you

Reply:
Good point. Let me fix. It's a much better API!